radixdlt-java/src/main/java/com/radixdlt/client/application/identity/Data.java

```diff
@@ -78,13 +78,17 @@ public Data build() {
 				bytes = this.bytes;
 			} else {
 				if (encryptorBuilder.getNumReaders() == 0) {
-					throw new IllegalStateException("Must either be unencrypted or have atleast one reader.");
+					throw new IllegalStateException("Must either be unencrypted or have at least one reader.");
 				}
 
-				ECKeyPair sharedKey = ECKeyPairGenerator.newInstance().generateKeyPair();
+				ECKeyPair sharedKey = new ECKeyPair();
```

Comment:
I don't like this way of generating new key pairs. It's non-obvious that this constructor generates a new key-pair rather than using some default, which is an important detail that is lost in this change.

Reply:
At the least, we should abandon the blank ECKeyPair constructor in favour of a static ECKeyPair.generateKeyPair method.

Reply:
Why? What would a "default keypair be"? Why should we treat this class different than any other POJO?

Reply:
Because it's not any other POJO. My point is that a default keypair doesn't make much sense, hence this constructor shouldn't exist (since it's the *default* constructor). Having `new ECKeyPair().equals(new ECKeyPair()) == false` is non-obvious and unexpected, which is bad.

Reply:
Aha ok, I see what you mean, will change to static!